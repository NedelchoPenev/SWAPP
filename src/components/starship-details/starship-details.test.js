import React from 'react';
import { shallow } from 'enzyme';

import StarshipDetails from './starship-details.component';

describe('StarshipDetails component', () => {
  let wrapper;
  const mockName = 'Jedi starfighter';
  const imageUrl = 'www.testImage.com';
  const mockClass = 'starfighter';
  const mockCost = '180000';
  const mockMaxAtmosphericSpeed = '1150';
  const mockHyperdriveRating = '1';
  const mockCrew = '10';
  let mockProps

  beforeEach(() => {
    mockProps = {
      starship: {
        name: mockName,
        image: imageUrl,
        starshipClass: mockClass,
        cost: mockCost,
        maxAtmosphericSpeed: mockMaxAtmosphericSpeed,
        hyperdriveRating: mockHyperdriveRating,
        crew: mockCrew,
      }
    };

    wrapper = shallow(<StarshipDetails {...mockProps} />);
  })

  it('should render StarshipDetails component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render starshipClass if starshipClass prop is available', () => {

    expect(wrapper.find('#pointer').length).toBe(5);

    expect(wrapper.find('#content').first().text()).toEqual(mockClass);
  });

  it('should NOT render if starshipClass is NOT available', () => {
    mockProps = {
      starship: {
        name: mockName,
        image: imageUrl,
        starshipClass: null,
        cost: mockCost,
        maxAtmosphericSpeed: mockMaxAtmosphericSpeed,
        hyperdriveRating: mockHyperdriveRating,
        crew: mockCrew,
      }
    };

    wrapper.setProps(mockProps)

    expect(wrapper.find('#pointer').length).toBe(4);

    expect(wrapper.find('#content').first().text()).not.toEqual(mockClass);
  });

  it('should render cost if cost prop is available', () => {

    expect(wrapper.find('#pointer').length).toBe(5);

    expect(wrapper.find('#content').at(1).text()).toEqual(`${mockCost} credits`);
  });

  it('should render 0 credits if cost is NOT available', () => {
    mockProps = {
      starship: {
        name: mockName,
        image: imageUrl,
        starshipClass: mockClass,
        cost: null,
        maxAtmosphericSpeed: mockMaxAtmosphericSpeed,
        hyperdriveRating: mockHyperdriveRating,
        crew: mockCrew,
      }
    };

    wrapper.setProps(mockProps)

    expect(wrapper.find('#content').at(1).text()).toEqual('0 credits');
  });

  it('should render crew if crew prop is available', () => {

    expect(wrapper.find('#pointer').length).toBe(5);

    expect(wrapper.find('#content').at(2).text()).toEqual(mockCrew);
  });

  it('should NOT render if crew is NOT available', () => {
    mockProps = {
      starship: {
        name: mockName,
        image: imageUrl,
        starshipClass: mockClass,
        cost: mockCost,
        maxAtmosphericSpeed: mockMaxAtmosphericSpeed,
        hyperdriveRating: mockHyperdriveRating,
        crew: null,
      }
    };

    wrapper.setProps(mockProps)

    expect(wrapper.find('#pointer').length).toBe(4);

    expect(wrapper.find('#content').at(2).text()).not.toEqual(mockCrew);
  });

  it('should render maxAtmosphericSpeed if maxAtmosphericSpeed prop is available', () => {

    expect(wrapper.find('#pointer').length).toBe(5);

    expect(wrapper.find('#content').at(3).text()).toEqual(mockMaxAtmosphericSpeed);
  });

  it('should NOT render if maxAtmosphericSpeed is NOT available', () => {
    mockProps = {
      starship: {
        name: mockName,
        image: imageUrl,
        starshipClass: mockClass,
        cost: mockCost,
        maxAtmosphericSpeed: null,
        hyperdriveRating: mockHyperdriveRating,
        crew: mockCrew,
      }
    };

    wrapper.setProps(mockProps)

    expect(wrapper.find('#pointer').length).toBe(4);

    expect(wrapper.find('#content').at(3).text()).not.toEqual(mockMaxAtmosphericSpeed);
  });

  it('should render hyperdriveRating if hyperdriveRating prop is available', () => {

    expect(wrapper.find('#pointer').length).toBe(5);

    expect(wrapper.find('#content').at(4).text()).toEqual(mockHyperdriveRating);
  });

  it('should NOT render if hyperdriveRating is NOT available', () => {
    mockProps = {
      starship: {
        name: mockName,
        image: imageUrl,
        starshipClass: mockClass,
        cost: mockCost,
        maxAtmosphericSpeed: mockMaxAtmosphericSpeed,
        hyperdriveRating: null,
        crew: mockCrew,
      }
    };

    wrapper.setProps(mockProps)

    expect(wrapper.find('#pointer').length).toBe(4);

    expect(wrapper.find('#content').at(4)).toEqual({});
  });
});