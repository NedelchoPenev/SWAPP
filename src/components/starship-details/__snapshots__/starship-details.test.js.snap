// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarshipDetails component should render StarshipDetails component 1`] = `
<styled.div>
  <h2>
    Jedi starfighter
  </h2>
  <img
    alt="Jedi starfighter"
    src="www.testImage.com"
  />
  <p>
    <span
      id="pointer"
    >
      Class: 
    </span>
    <span
      id="content"
    >
      starfighter
    </span>
  </p>
  <p>
    <span
      id="pointer"
    >
      Cost: 
    </span>
    <span
      id="content"
    >
      180000
       credits
    </span>
  </p>
  <p>
    <span
      id="pointer"
    >
      Crew: 
    </span>
    <span
      id="content"
    >
      10
    </span>
  </p>
  <p>
    <span
      id="pointer"
    >
      Max Atmospheric Speed: 
    </span>
    <span
      id="content"
    >
      1150
    </span>
  </p>
  <p>
    <span
      id="pointer"
    >
      Hyperdrive Rating: 
    </span>
    <span
      id="content"
    >
      1
    </span>
  </p>
</styled.div>
`;
